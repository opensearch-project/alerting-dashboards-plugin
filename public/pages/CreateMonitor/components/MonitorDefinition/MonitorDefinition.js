/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import FormikSelect from '../../../../components/FormControls/FormikSelect/FormikSelect';
import { OS_AD_PLUGIN } from '../../../../utils/constants';

const defaultSelectDefinitions = [
  { value: 'graph', text: 'Define using visual graph' },
  { value: 'query', text: 'Define using extraction query' },
];

const bucketLevelMonitorDefinitions = [
  { value: 'graph', text: 'Define using visual graph' },
  { value: 'query', text: 'Define using extraction query' },
];

const onChangeDefinition = (e, form, resetResponse) => {
  const type = e.target.value;
  resetResponse();
  form.setFieldValue('searchType', type);
};

const selectDefinitions = (plugins, isBucketLevelMonitor) => {
  const definitionOptions = isBucketLevelMonitor
    ? bucketLevelMonitorDefinitions
    : defaultSelectDefinitions;
  return plugins === undefined || plugins.indexOf(OS_AD_PLUGIN) == -1
    ? definitionOptions
    : [...definitionOptions, { value: 'ad', text: 'Define using anomaly detector' }];
};

const MonitorDefinition = ({ resetResponse, plugins, isBucketLevelMonitor }) => (
  <div>
    <FormikSelect
      name="searchType"
      formRow
      rowProps={{
        label: 'Method of definition',
        style: { paddingLeft: '10px' },
      }}
      inputProps={{
        options: selectDefinitions(plugins, isBucketLevelMonitor),
        onChange: (e, field, form) => {
          onChangeDefinition(e, form, resetResponse);
        },
      }}
    />
  </div>
);

export default MonitorDefinition;
