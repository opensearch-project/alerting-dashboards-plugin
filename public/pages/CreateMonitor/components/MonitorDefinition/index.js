/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import MonitorDefinition from './MonitorDefinition';

export default MonitorDefinition;
