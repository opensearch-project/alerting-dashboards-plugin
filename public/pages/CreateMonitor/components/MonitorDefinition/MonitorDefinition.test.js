/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from 'enzyme';
import { Formik } from 'formik';

import MonitorDefinition from './MonitorDefinition';

describe('MonitorDefinition', () => {
  test('renders', () => {
    const component = (
      <Formik>
        <MonitorDefinition resetResponse={() => {}} />
      </Formik>
    );
    expect(render(component)).toMatchSnapshot();
  });
});
