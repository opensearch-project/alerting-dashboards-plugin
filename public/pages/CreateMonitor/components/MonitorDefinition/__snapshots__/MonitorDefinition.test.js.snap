// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonitorDefinition renders 1`] = `
<div>
  <div
    class="euiFormRow"
    id="searchType-form-row-row"
    style="padding-left:10px"
  >
    <div
      class="euiFormRow__labelWrapper"
    >
      <label
        class="euiFormLabel euiFormRow__label"
        for="searchType-form-row"
      >
        Method of definition
      </label>
    </div>
    <div
      class="euiFormRow__fieldWrapper"
    >
      <div
        class="euiFormControlLayout"
      >
        <div
          class="euiFormControlLayout__childrenWrapper"
        >
          <select
            class="euiSelect"
            id="searchType"
            name="searchType"
          >
            <option
              value="graph"
            >
              Define using visual graph
            </option>
            <option
              value="query"
            >
              Define using extraction query
            </option>
          </select>
          <div
            class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
          >
            <span
              class="euiFormControlLayoutCustomIcon"
            >
              <svg
                aria-hidden="true"
                class="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
